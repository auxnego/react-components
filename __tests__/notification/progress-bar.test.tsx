import React from "react"
import {create} from "react-test-renderer";

import ProgressBar from "../../notification/progress-bar";

it("Indeterminated progress bar", () => {

  expect(create(
    <ProgressBar indeterminate />
  ).toJSON()).toMatchSnapshot();

});

it("With total and current=0", () => {

  expect(create(
    <ProgressBar total={124} />
  ).toJSON()).toMatchSnapshot();

});

it("with total=200 and current=100", () => {

  expect(create(
    <ProgressBar current={100} total={200}/>
  ).toJSON()).toMatchSnapshot();

});

it(" with total=200 and current=200 and indicatorFn", () => {

  expect(create(
    <ProgressBar total={200} current={200} indicatorFn={(current, total, indeterminate) => {
      return <i>{current} / {total}</i>;
    }}/>
  ).toJSON()).toMatchSnapshot();

});
