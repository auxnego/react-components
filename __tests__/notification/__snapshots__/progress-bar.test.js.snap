// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` with total=200 and current=200 and indicatorFn 1`] = `
<div
  className="progress-bar__root"
>
  <i>
    200
     / 
    200
  </i>
  <div
    className="progress-bar__bar"
  >
    <div
      className="progress-bar__indicator progress-bar__indicator--100"
    />
  </div>
</div>
`;

exports[`Indeterminated progress bar 1`] = `
<div
  className="progress-bar__root progress-bar__root--indeterminate"
>
  <div
    className="progress-bar__bar"
  >
    <div
      className="progress-bar__indicator"
    />
  </div>
</div>
`;

exports[`With total and current=0 1`] = `
<div
  className="progress-bar__root"
>
  <div
    className="progress-bar__bar"
  >
    <div
      className="progress-bar__indicator progress-bar__indicator--0"
    />
  </div>
</div>
`;

exports[`with total=200 and current=100 1`] = `
<div
  className="progress-bar__root"
>
  <div
    className="progress-bar__bar"
  >
    <div
      className="progress-bar__indicator progress-bar__indicator--50"
    />
  </div>
</div>
`;
